### ISSUE TITLE
[Bug] Can not use default devserver's port

### ISSUE THREAD

**Issue by heavenshell** [react-static/react-static#1285]
### Reporting a bug?

  Provide as much information as you can to reproduce the issue.
  **If the issue is not reproducible, it can't be fixed!**

### Environment

Run and copy the result of:

`npx envinfo --system --npmPackages react* --binaries --npmGlobalPackages react* --browsers` 

here:

```


  System:
    OS: macOS 10.14.6
    CPU: (8) x64 Intel(R) Core(TM) i7-3720QM CPU @ 2.60GHz
    Memory: 432.81 MB / 16.00 GB
    Shell: 5.7.1 - /opt/local/bin/zsh
  Binaries:
    Node: 10.16.0 - /tmp/nodebrew/current/bin/node
    Yarn: 1.17.3 - /opt/local/bin/yarn
    npm: 6.9.0 - /tmp/nodebrew/current/bin/npm
  Browsers:
    Chrome: 76.0.3809.100
    Firefox: 68.0.2
    Safari: 12.1.2
  npmPackages:
    react: ^16.8.2 => 16.9.0 
    react-dom: ^16.8.2 => 16.9.0 
    react-static: ^7.0.0 => 7.1.0 
    react-static-plugin-reach-router: latest => 7.0.10 
    react-static-plugin-sitemap: latest => 7.0.10 
    react-static-plugin-source-filesystem: latest => 7.1.0 
    react-static-plugin-typescript: latest => 7.0.10 
```

of course include any other package versions here if relevant. 

### Steps to Reproduce the problem

**Base your steps off of any freshly installed react-static template!**

1.  `npx react-static create --template typescript --name=my-static-site`
2.  `cd my-static-site`
3.  `yarn`
4.  `yarn upgrade`
5.  `yarn start`
6.  Display `Warning! Port 3000 is not available. Using port 3488 instead!`

### Expected Behavior

`Port 3000` does not use, so `react-static` shouold use port 3000.

**Comment by heavenshell** [comment]
Also, I added `devServer` option to static.site.js, this does not work.

```
export default {
  devServer: {
    port: 9999,
  },
```

```
$ yarn start
yarn run v1.17.3
$ react-static start
Starting Development Server...
Building Routes...
Importing routes from directory...
[✓] Routes Built (0.2s)
Building Templates...
[✓] Templates Built
Warning! Port 9999 is not available. Using port 10613 instead!
Bundling Application...
Starting type checking service...
Using 1 worker with 2048MB memory limit
Fetching Site Data...
[✓] Site Data Downloaded

  Replace Autoprefixer browsers option to Browserslist config.
  Use browserslist key in package.json or .browserslistrc file.

  Using browsers option cause some error. Browserslist config 
  can be used for Babel, Autoprefixer, postcss-normalize and other tools.

  If you really need to use option, rename it to overrideBrowserslist.

  Learn more at:
  https://github.com/browserslist/browserslist#readme
  https://twitter.com/browserslist


[✓] Application Bundled (5.2s)
[✓] App serving at localhost:10613
```
Expected port is `9999`.

**Comment by heavenshell** [comment]
IMO `portfinder 1.0.22` is affected...

```
  "resolutions": {
    "portfinder": "1.0.21"
  },
```
If I fixed portfinder version as 1.0.21, works fine.

**Comment by SleeplessByte** [comment]
I can't reproduce this. If the resolution works for you, I will close this for now until we hit this more often.
